executor/src/witgen/global_constraints.rs

```diff
@@ -75,17 +75,27 @@ pub fn determine_global_constraints<'a, T: FieldElement>(
         .push(identity);
     }
 
-    log::debug!("Determined the following global range constraints:");
-    for (name, con) in &known_constraints {
-        log::debug!("  {name}: {con}");
-    }
+    log::debug!("Determined the following global range constraints on witness columns:");
+    let known_witness_constraints = known_constraints
+        .into_iter()
+        .filter_map(|(poly, cons)| {
+            assert!(!poly.next);
```

Comment:
I think that was also @georgwiese 's idea in the first place, wasn't it?

Reply:
(not a blocker for this PR)

Reply:
Hmm, few thoughts on this:
- I didn't dive into the code that computes global range constraints here, but it's kinda weird that the other functions work on Polynomial references instead of `PolyID`. For example, I guess defining a bit constraint as `X' * (1 - X') = 0` would break the code. What you'd *really* need to do is to detect range constraints per reference, group by ID and intersect the ranges.
- What's nice about a `PolynomialReference` is that it can pretty-print itself because it has (i.e. owns) the name of the polynomial. Because of that, we use `&PolynomialReference` as keys in various places...

I think what would be nice is to have something like:
```
struct Polynomial:
  name: &'static str,
  id: PolyID
```

or

```
struct Polynomial<'a>:
  name: &'a str,
  id: PolyID
```

The first approach involves leaking the column names, which I think would be fine and prevent using convoluting the code with lifetime annotations...

What do you think?